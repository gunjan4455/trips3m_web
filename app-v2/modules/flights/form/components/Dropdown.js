import React, { Component } from 'react';
import PropTypes from 'prop-types';
import debounce from 'lodash/debounce';

import styled from '@emotion/styled';

const SEARCH_MINIMUM_LENGTH = 3;
const SEARCH_DEBOUNCE_TIME = 250;

const GDropDownListDiv = styled.div({
  maxHeight: '205px',
  overflowY: 'auto'
});

class Dropdown extends Component {

  constructor(props) {
    super(props);
    this.state = { isOpen: false, inputValue: '', items: props.options };
    this.loadOptions = debounce(this.searchTerm, SEARCH_DEBOUNCE_TIME);
  }

  componentDidMount() {
    document.addEventListener('click', this.handleClickOutside);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClickOutside);
  }

  toggleMenu = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  handleSelection = (e) => {
    const { input } = this.props;
    this.toggleMenu();
    input.onChange(e.target.getAttribute('data-val'));
  };

  searchTerm = () => {
    const { inputValue } = this.state;
    const { getOptions, options } = this.props;
    if (
      inputValue &&
      inputValue.length >= SEARCH_MINIMUM_LENGTH &&
      getOptions
    ) {
      return getOptions(inputValue)
        .then((res) => this.setState({items: res.data.results }));
    }
    return this.setState({ items: options });
  };

  handleChange = (e) => {
    const {  input, fetchSuggestions } = this.props;
    const val = e.target.value;
    input.onChange(val);
    this.setState({ inputValue: val, isOpen: true }, () => {
      if (fetchSuggestions) {
        this.loadOptions();
      }
    });
  };

  handleClickOutside = event => {
    if (!(this.listElement_ && this.listElement_.contains && this.listElement_.contains(event.target))
      && !(this.inputElement_ && this.inputElement_.contains && this.inputElement_.contains(event.target))
    ) {
      this.setState({ isOpen: false });
    }
  };

  handleClick = () => {
    const { trackSegment } = this.props;
    this.toggleMenu();
    trackSegment();
  };

  render() {
    const { input, options, meta: { touched , error }, disabled, placeholder, readOnly, inputClassName } = this.props;
    const { isOpen, inputValue, items } = this.state;
    const newItems = (input.value || inputValue) ? items : options;

    return (
      <div className={`relative pb15 ${touched && (error && 'mb8')}`}>
        <input
          placeholder={placeholder}
          value={input.value || inputValue}
          className={`${inputClassName} text-capitalize ${'select'} ${touched && (error && 'input-error mb2')}`}
          disabled={disabled}
          onChange={this.handleChange}
          onClick={this.handleClick}
          readOnly={readOnly}
          ref={(ref) => this.inputElement_ = ref}
        />
        { isOpen ?
          <GDropDownListDiv className="border sbcw absolute z10 l0 r0 showScroll" ref={(ref) => this.listElement_ = ref}>
            {
              newItems.map(option => (
                <div className="p8 bb" data-val={option.value} key={`isd_${option.value}`} onClick={this.handleSelection}>
                  {option.label}
                </div>
              ))
            }
          </GDropDownListDiv>
        : null}
        {
          touched &&
          (error && <span className="input-error-text absolute b0

          ">{error}</span>)
        }
      </div>
    );
  }
}

Dropdown.propTypes = {
  options: PropTypes.array.isRequired,
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  getOptions: PropTypes.func,
  trackSegment: PropTypes.func,
  fetchSuggestions: PropTypes.bool,
  readOnly: PropTypes.bool,
  inputClassName: PropTypes.string
};

Dropdown.defaultProps = {
  disabled: false,
  placeholder: '',
  trackSegment: () => {},
  readOnly: false,
  fetchSuggestions: true,
  inputClassName: 'pl48'
};

export default Dropdown;
