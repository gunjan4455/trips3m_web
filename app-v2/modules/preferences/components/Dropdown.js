import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = ({ disabled, input, defaultValue, options, meta: { touched, error } }) => (
  <div>
    <select
      {...input}
      onBlur={() => input.onBlur(input.value)}
      className={`${'select'} ${touched && (error && 'input-error')}`}
      defaultValue={defaultValue}
      disabled={disabled}
    >
      {
        options.map(option => (
          <option value={option.value} key={`isd_${option.value}`}>
            {option.label}
          </option>
        ))
      }
    </select>
    {
      touched &&
      (error && <span className="input-error-text">{error}</span>)
    }
  </div>
);

Dropdown.propTypes = {
  options: PropTypes.array,
  defaultValue: PropTypes.any,
  input: PropTypes.object,
  meta: PropTypes.object,
  disabled: PropTypes.bool
};

Dropdown.defaultProps = {};

export default Dropdown;
