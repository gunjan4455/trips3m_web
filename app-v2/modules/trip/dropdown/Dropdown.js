import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { rtripPath } from 'helpers/urlHelpers';
import { DownArrow } from 'helpers/Icon/Icon';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import { rescue } from 'helpers/utils';
import {
  GTripListDiv, GDropdownButton, GTripsListUl, GArrowDownSpan
} from './G';

class Dropdown extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      dropDownOpen: false
    };
    this.getTrips = this.getTrips.bind(this);
    this.toggleDropDown = this.toggleDropDown.bind(this);
    this.handleClickOutside = this.handleClickOutside.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    document.addEventListener('click', this.handleClickOutside);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClickOutside);
  }

  handleClickOutside = event => {
    if (this.state.dropDownOpen && !this.tripListElement_.contains(event.target)) {
      this.setState({ dropDownOpen: false });
    }
  };

  trackSegmentEvent = (event, section, object, cta) => {
    const { trackSegment } = this.props;

    rescue(() => {
      trackSegment({
        event,
        section,
        object,
        cta
      });
    });
  };

  handleChange = (trip) => {
    this.trackSegmentEvent(ORGANISM_CLICKED, 'Trips dropdown section', '', trip.firstDestinationName);
    this.setState({ dropDownOpen: !this.state.dropDownOpen });
  };

  toggleDropDown = () => {
    this.setState({ dropDownOpen: !this.state.dropDownOpen });
  };

  getTrips = () => {
    const { trips } = this.props;
    return (trips.allIds.map(tripId => {
      const trip = trips.byId[tripId];
      const rTripUrl = rtripPath(trip.id);

      return (
        <li key={trip.id}>
          <Link to={rTripUrl} onClick={() => this.handleChange(trip)}>
            <span title={trip.firstDestinationName}
                  className="block ellipsis f14 at_DestinationName">{trip.firstDestinationName || trip.toLoc}</span>
            <span className="block f12 pfc1 at_bookingId">Booking ID #{trip.id}</span>
          </Link>
        </li>
      );
    }));
  };

  render() {
    const { trips, currentTripId } = this.props;
    const currentTrip = trips.byId[currentTripId];

    return (
      <GTripListDiv className="flexFull relative sbcw radius2">
        <div className="clearfix relative">
          <div ref={(ref) => this.tripListElement_ = ref}>
            <GDropdownButton className="radius2 at_tripListButton relative pr32 ellipsis" onClick={this.toggleDropDown}>
              <span
                className="fw7 at_destinationname">{currentTrip.firstDestinationName || currentTrip.toLoc}</span>
              {
                <GArrowDownSpan className={!this.state.dropDownOpen ? '' : 'arrowDown'}>
                  <DownArrow />
                </GArrowDownSpan>
              }
            </GDropdownButton>
            <GTripsListUl
              className={`dropDownList radius2 bs4 mt3 sbcw ${!this.state.dropDownOpen ? ' dropdownClosed' : ''}`}>
              {
                this.props.trips.allIds && this.props.trips.allIds.length > 0
                  ? this.getTrips()
                  : null
              }
            </GTripsListUl>
          </div>
        </div>
      </GTripListDiv>
    );
  }
}

Dropdown.propTypes = {
  trips: PropTypes.object.isRequired,
  currentTripId: PropTypes.number.isRequired,
  trackSegment: PropTypes.func.isRequired,
};

export default Dropdown;
