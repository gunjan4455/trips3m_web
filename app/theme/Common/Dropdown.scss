.Dropdown-root {
    position: absolute;
    z-index: 3;
    cursor: pointer;
    width: 100%;
    top: 50%;
    transform: translateY(-50%);
    height: 100%;
}

.Dropdown-control {
  position: relative;
  overflow: hidden;
  border-radius: 2px;
  box-sizing: border-box;
  color: #3e3e3e;
  outline: none;
  padding: 8px 10px 8px 10px;
  transition: all 200ms ease;
  line-height: 1;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  -o-transform: translateY(-50%);
  transform: translateY(-50%);
}

.Dropdown-menu {
  background-color: #fff;
  border: 1px solid #B2B2B2;
  box-sizing: border-box;
  margin-top: 1px;
  max-height: 200px;
  overflow-y: auto;
  position: absolute;
  top: 100%;
  left:0px;
  width: 100%;
  z-index: 100;
  -webkit-overflow-scrolling: touch;
}

.Dropdown-menu .Dropdown-group > .Dropdown-title{
  padding: 8px 10px;
  color: rgba(51, 51, 51, 1);
  font-weight: bold;
  text-transform: capitalize;
}

.Dropdown-option {
  box-sizing: border-box;
  color: #3e3e3e;
  cursor: pointer;
  display: block;
  padding: 8px 10px;
  border-bottom: 1px solid #f0f0f0;
}

.Dropdown-option:last-child {
  border-bottom-right-radius: 2px;
   border-bottom-left-radius: 2px;
}

.Dropdown-option:hover {
  background-color: #20A397;
  color: #fff;
}

.Dropdown-noresults {
  box-sizing: border-box;
  color: #3e3e3e;
  cursor: default;
  display: block;
  padding: 8px 10px;
}

.input-dropdown-box {
  height: 40px;
  position: relative;
  cursor: pointer;
}
.input-dropdown-box {
  height: 40px;
  cursor: pointer;
  background-color: #ffffff;
  border: 1px solid #B2B2B2;
}
.input-dropdown-box.input-dropdown-box-small {
  height: 32px;
}
.input-dropdown-box.input-dropdown-box-long {
  height: 48px;
}

.input-with-icon-box .Dropdown-control {
	padding-left: 45px;
}
