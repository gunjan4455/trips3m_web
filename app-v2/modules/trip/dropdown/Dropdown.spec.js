import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Dropdown from './Dropdown';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  trips: {
    allIds: [],
    byId: {
      '1111': {
        id: '1111',
        firstDestinationName: 'Kerala',
        toLoc: 'Kerala'
      },
      '2222': {
        id: '2222',
        firstDestinationName: 'Himachal',
        toLoc: 'Kerala'
      },
      '3333': {
        id: '3333',
        firstDestinationName: null,
        toLoc: 'pakistan'
      }
    }
  },
  currentTripId: '2222',
  trackSegment: jest.fn,
  push: jest.fn
};

const shallowWrapperTripList = (props) => {
  return shallow(<Dropdown {...props}/>);
};

describe('Trip List: Render Correctly ', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrapperTripList(props);

    expect(wrapper).toMatchSnapshot();
  });

  it('should show toLoc when firstDestinationName is null', () => {
    props.currentTripId = '3333';
    const wrapper = shallowWrapperTripList(props);

    expect(wrapper.find('.at_destinationname').text()).toEqual('pakistan');
  });
});
