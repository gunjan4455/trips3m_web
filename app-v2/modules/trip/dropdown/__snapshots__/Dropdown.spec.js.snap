// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip List: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    currentTripId="2222"
    push={[Function]}
    trackSegment={[Function]}
    trips={
      Object {
        "allIds": Array [],
        "byId": Object {
          "1111": Object {
            "firstDestinationName": "Kerala",
            "id": "1111",
            "toLoc": "Kerala",
          },
          "2222": Object {
            "firstDestinationName": "Himachal",
            "id": "2222",
            "toLoc": "Kerala",
          },
          "3333": Object {
            "firstDestinationName": null,
            "id": "3333",
            "toLoc": "pakistan",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="clearfix relative"
      >
        <div>
          <ForwardRef(render)
            className="radius2 at_tripListButton relative pr32 ellipsis"
            onClick={[Function]}
          >
            <span
              className="fw7 at_destinationname"
            >
              Himachal
            </span>
            <ForwardRef(render)
              className=""
            >
              <IconSvg />
            </ForwardRef(render)>
          </ForwardRef(render)>
          <ForwardRef(render)
            className="dropDownList radius2 bs4 mt3 sbcw  dropdownClosed"
          />
        </div>
      </div>,
      "className": "flexFull relative sbcw radius2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <ForwardRef(render)
            className="radius2 at_tripListButton relative pr32 ellipsis"
            onClick={[Function]}
          >
            <span
              className="fw7 at_destinationname"
            >
              Himachal
            </span>
            <ForwardRef(render)
              className=""
            >
              <IconSvg />
            </ForwardRef(render)>
          </ForwardRef(render)>
          <ForwardRef(render)
            className="dropDownList radius2 bs4 mt3 sbcw  dropdownClosed"
          />
        </div>,
        "className": "clearfix relative",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef(render)
              className="radius2 at_tripListButton relative pr32 ellipsis"
              onClick={[Function]}
            >
              <span
                className="fw7 at_destinationname"
              >
                Himachal
              </span>
              <ForwardRef(render)
                className=""
              >
                <IconSvg />
              </ForwardRef(render)>
            </ForwardRef(render)>,
            <ForwardRef(render)
              className="dropDownList radius2 bs4 mt3 sbcw  dropdownClosed"
            />,
          ],
        },
        "ref": [Function],
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span
                  className="fw7 at_destinationname"
                >
                  Himachal
                </span>,
                <ForwardRef(render)
                  className=""
                >
                  <IconSvg />
                </ForwardRef(render)>,
              ],
              "className": "radius2 at_tripListButton relative pr32 ellipsis",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Himachal",
                  "className": "fw7 at_destinationname",
                },
                "ref": null,
                "rendered": "Himachal",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <IconSvg />,
                  "className": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "span",
                  "__emotion_forwardProp": undefined,
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "& svg": Object {
                        "height": "6px",
                        "width": "10px",
                      },
                      "display": "block",
                      "height": "6px",
                      "lineHeight": "0",
                      "position": "absolute",
                      "right": "15px",
                      "top": "50%",
                      "transform": "translateY(-50%)",
                      "transition": "transform 1s",
                      "width": "10px",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(span)",
                  "render": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "button",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                Object {
                  "height": "40px",
                  "padding": "0 15px",
                  "textAlign": "left",
                  "width": "100%",
                },
              ],
              "defaultProps": undefined,
              "displayName": "Styled(button)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "className": "dropDownList radius2 bs4 mt3 sbcw  dropdownClosed",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "ul",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                Object {
                  "& li": Object {
                    "& a": Object {
                      "& p": Object {
                        "margin": "0",
                      },
                      "&:hover, &:active, &:focus": Object {
                        "background": "#009688",
                        "color": "#fff",
                        "textDecoration": "none",
                      },
                      "color": "#3e3e3e",
                      "display": "inline-block",
                      "padding": "8px 15px",
                      "width": "100%",
                    },
                    "display": "block",
                    "listStyle": "none",
                  },
                  "& li.noTrips": Object {
                    "padding": "15px",
                  },
                  "margin": "0",
                  "padding": "0",
                  "position": "absolute",
                  "width": "100%",
                  "zIndex": "101",
                },
              ],
              "defaultProps": undefined,
              "displayName": "Styled(ul)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        Object {
          "& .arrowDown": Object {
            "transform": "rotate(180deg)",
          },
          "& .dropDownList": Object {
            "borderTop": "0 none",
            "maxHeight": "160px",
            "overflowX": "hidden",
            "overflowY": "auto",
            "transition": "max-height 0.2s",
          },
          "& .dropdownClosed": Object {
            "border": "0 none",
            "maxHeight": "0",
          },
          "& select": Object {
            "appearance": "none",
            "bottom": "0",
            "left": "0",
            "opacity": "1",
            "position": "absolute",
            "right": "0",
            "top": "0",
            "width": "100%",
            "zIndex": "1",
          },
          "minHeight": "40px",
        },
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="clearfix relative"
        >
          <div>
            <ForwardRef(render)
              className="radius2 at_tripListButton relative pr32 ellipsis"
              onClick={[Function]}
            >
              <span
                className="fw7 at_destinationname"
              >
                Himachal
              </span>
              <ForwardRef(render)
                className=""
              >
                <IconSvg />
              </ForwardRef(render)>
            </ForwardRef(render)>
            <ForwardRef(render)
              className="dropDownList radius2 bs4 mt3 sbcw  dropdownClosed"
            />
          </div>
        </div>,
        "className": "flexFull relative sbcw radius2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <ForwardRef(render)
              className="radius2 at_tripListButton relative pr32 ellipsis"
              onClick={[Function]}
            >
              <span
                className="fw7 at_destinationname"
              >
                Himachal
              </span>
              <ForwardRef(render)
                className=""
              >
                <IconSvg />
              </ForwardRef(render)>
            </ForwardRef(render)>
            <ForwardRef(render)
              className="dropDownList radius2 bs4 mt3 sbcw  dropdownClosed"
            />
          </div>,
          "className": "clearfix relative",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef(render)
                className="radius2 at_tripListButton relative pr32 ellipsis"
                onClick={[Function]}
              >
                <span
                  className="fw7 at_destinationname"
                >
                  Himachal
                </span>
                <ForwardRef(render)
                  className=""
                >
                  <IconSvg />
                </ForwardRef(render)>
              </ForwardRef(render)>,
              <ForwardRef(render)
                className="dropDownList radius2 bs4 mt3 sbcw  dropdownClosed"
              />,
            ],
          },
          "ref": [Function],
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span
                    className="fw7 at_destinationname"
                  >
                    Himachal
                  </span>,
                  <ForwardRef(render)
                    className=""
                  >
                    <IconSvg />
                  </ForwardRef(render)>,
                ],
                "className": "radius2 at_tripListButton relative pr32 ellipsis",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Himachal",
                    "className": "fw7 at_destinationname",
                  },
                  "ref": null,
                  "rendered": "Himachal",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <IconSvg />,
                    "className": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "__emotion_base": "span",
                    "__emotion_forwardProp": undefined,
                    "__emotion_real": [Circular],
                    "__emotion_styles": Array [
                      Object {
                        "& svg": Object {
                          "height": "6px",
                          "width": "10px",
                        },
                        "display": "block",
                        "height": "6px",
                        "lineHeight": "0",
                        "position": "absolute",
                        "right": "15px",
                        "top": "50%",
                        "transform": "translateY(-50%)",
                        "transition": "transform 1s",
                        "width": "10px",
                      },
                    ],
                    "defaultProps": undefined,
                    "displayName": "Styled(span)",
                    "render": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "button",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "height": "40px",
                    "padding": "0 15px",
                    "textAlign": "left",
                    "width": "100%",
                  },
                ],
                "defaultProps": undefined,
                "displayName": "Styled(button)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "className": "dropDownList radius2 bs4 mt3 sbcw  dropdownClosed",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "ul",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "& li": Object {
                      "& a": Object {
                        "& p": Object {
                          "margin": "0",
                        },
                        "&:hover, &:active, &:focus": Object {
                          "background": "#009688",
                          "color": "#fff",
                          "textDecoration": "none",
                        },
                        "color": "#3e3e3e",
                        "display": "inline-block",
                        "padding": "8px 15px",
                        "width": "100%",
                      },
                      "display": "block",
                      "listStyle": "none",
                    },
                    "& li.noTrips": Object {
                      "padding": "15px",
                    },
                    "margin": "0",
                    "padding": "0",
                    "position": "absolute",
                    "width": "100%",
                    "zIndex": "101",
                  },
                ],
                "defaultProps": undefined,
                "displayName": "Styled(ul)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          Object {
            "& .arrowDown": Object {
              "transform": "rotate(180deg)",
            },
            "& .dropDownList": Object {
              "borderTop": "0 none",
              "maxHeight": "160px",
              "overflowX": "hidden",
              "overflowY": "auto",
              "transition": "max-height 0.2s",
            },
            "& .dropdownClosed": Object {
              "border": "0 none",
              "maxHeight": "0",
            },
            "& select": Object {
              "appearance": "none",
              "bottom": "0",
              "left": "0",
              "opacity": "1",
              "position": "absolute",
              "right": "0",
              "top": "0",
              "width": "100%",
              "zIndex": "1",
            },
            "minHeight": "40px",
          },
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
